rfc/2020/alertmanager-integration/christyjacob4.md

```diff
@@ -0,0 +1,503 @@
+
+* Once this RFC is implemented, users will have the option to install **Alertmanager** in the control plane using the `addon-config` flag in `linkerd install`
+
+
+- The optional **Alertmanager** installation will come with some default alerts
+  which users can configure to appropriate channels like slack, email, pagerduty, wechat etc.
+
+- Users will also be able to specify custom rules for Prometheus as part of the configuration for `--addon-config` in `config.yaml`.
+
+- Integration with Service Profiles will allow users to define rules and alerts for per route metrics providing extremely fine grain alerting mechanisms for their apps.
+
+A POC for this has been done in [linkerd/linkerd2#1726](https://github.com/linkerd/linkerd2/pull/4124)
+
+# Design proposal (Step 2)
+
+The Integration of Alertmanager with the Existing Control Plane is a multi step process. It can be divided into the following stages
+
+---
+- ## Providing an option to include AlertManager in the installation
+The *Linkerd* install command is currently used to generate a helm chart based on various flags and options that are passed to it. The helm chart is then applied using `kubectl apply` .
+With the recent addition of [**Tracing**](https://github.com/linkerd/linkerd2/pull/3955) to *Linkerd* installation as an add-on, we have a framework to add more modules as part of the Control Plane installation.
+
+The plan is to pass a configuration file `config.yaml` in the `--addon-config` flag along with `linkerd install`, that would allow the user to install *Linkerd* with Alertmanager configured out-of-the-box, as an addon. So the complete command would look like this 
+
+```sh
+linkerd install --addon-config config.yaml | kubectl apply -f -
+```
+
+This will simply enable Alertmanager with a default, publicly available [**webhook receiver**](#webhook-configuration) and configure Prometheus with a few default rules.  
+
+> This will enable the user to 
+>  * Check if the default Prometheus rules are firing.  
+>  * Check if Alertmanager is able to receive alerts from Prometheus.
+>  * Validate if Alertmanager is able to forward alerts to receivers.
+
+Additional configuration for a more specific Alertmanager install can be specified in the [`config.yaml`](#installation-yaml) file. This would contain all the configurations required to setup a full feldged Alertmanager install like enabling the default webhook flag, path to Prometheus files, path to Alertmanager Receivers file etc. The documentation for the `config.yaml` file is given [**here**](#installation-yaml).
+
+For the Alertmanager to function properly, it should have the correct iptables and roles, this is done by the proxy injector and RBAC settings. More information about the implementation of the Alertmanager Helm chart can be found in this [**section**](#alertmanager-helm-chart). 
```

Comment:
I don't think there is an any significant reason to mention this w.r.t iptables rules right?. (Is there?) AlertManager is also like any other control-plane component which get's injected (without any specific configuration) right?

Reply:
Yes @Pothulapati  there is no explicit reason to mention this. Just mentioned it for the sake of completeness.